kvbc/src/merkle_tree_db_adapter.cpp

```diff
@@ -387,6 +397,14 @@ SetOfKeyValuePairs DBAdapter::lastReachableBlockDbUpdates(const SetOfKeyValuePai
 
   for (const auto &[k, v] : non_provable_kvs) {
     dbUpdates[DBKeyManipulator::genNonProvableBlockDbKey(blockId, k)] = v;
+    if (blockId) {
+      const auto &stale_kv = getValueForNonProvableKey(k, blockId - 1);
```

Comment:
Something that came up after discussions with @andrewjstone is that if we don't save non-provable keys in the block node, then we will need to seek in `getRawBlock()` instead of doing direct lookups for them. We have seen some potential performance issues with iterators and also`getRawBlock()` itself. What I am saying is we might consider handling non-provable similarly to provable ones, i.e. writing them in the block node.

Reply:
It was discussed offline, the implementation will stay the same for now.